packages/repo-metadata-lint/src/file-iterator.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {OctokitType} from './utils/octokit-util';
+import {Storage} from '@google-cloud/storage';
+import {logger} from 'gcf-utils';
+
+/**
+ * This interface is used for comforting typescript's type system to
+ * deal with the response from `octokit.repos.getContent`.
+ */
+interface File {
+  content: string | undefined;
+}
+
+/**
+ * This function is used for comforting typescript's type system to
+ * deal with the response from `octokit.repos.getContent`.
+ */
+function isFile(file: File | unknown): file is File {
+  return (file as File).content !== undefined;
+}
+
+// Load full contents of GitHub repository from commit cache,
+// use this to iterate over individual .repo-metadata.json files.
+// This approach is used for the benefit of mono-repos.
+export class FileIterator {
```

Comment:
TODO: open issue to raise issue if `.repo-metadata.json` is missing entirely.

Reply:
I think this edge-case should be impossible once we move towards automatically generating new libraries, so let's just skip this feature.